src/responder/secrets/local.c

```diff
@@ -372,14 +372,43 @@ int local_db_delete(TALLOC_CTX *mem_ctx,
                     struct local_context *lctx,
                     const char *req_path)
 {
+    TALLOC_CTX *tmp_ctx;
     struct ldb_dn *dn;
+    static const char *attrs[] = { NULL };
+    struct ldb_result *res;
     int ret;
 
+    tmp_ctx = talloc_new(mem_ctx);
+    if (!tmp_ctx) return ENOMEM;
+
     ret = local_db_dn(mem_ctx, lctx->ldb, req_path, &dn);
-    if (ret != EOK) return ret;
+    if (ret != EOK) goto done;
+
+    ret = ldb_search(lctx->ldb, tmp_ctx, &res, dn, LDB_SCOPE_BASE,
+                    attrs, LOCAL_CONTAINER_FILTER);
+    if (ret != EOK) goto done;
+
+    if (res->count == 1) {
+        ret = ldb_search(lctx->ldb, tmp_ctx, &res, dn, LDB_SCOPE_ONELEVEL,
+                         attrs, NULL);
+        if (ret != EOK) goto done;
+
+        if (res->count > 0) {
+            ret = EINVAL;
```

Comment:
Simo,

On Fri, Sep 9, 2016 at 3:09 PM, Simo Sorce <notifications@github.com> wrote:
>
> In src/responder/secrets/local.c:
>
> >      ret = local_db_dn(mem_ctx, lctx->ldb, req_path, &dn);
> > -    if (ret != EOK) return ret;
> > +    if (ret != EOK) goto done;
> > +
> > +    ret = ldb_search(lctx->ldb, tmp_ctx, &res, dn, LDB_SCOPE_BASE,
> > +                    attrs, LOCAL_CONTAINER_FILTER);
> > +    if (ret != EOK) goto done;
> > +
> > +    if (res->count == 1) {
> > +        ret = ldb_search(lctx->ldb, tmp_ctx, &res, dn, LDB_SCOPE_ONELEVEL,
> > +                         attrs, NULL);
> > +        if (ret != EOK) goto done;
> > +
> > +        if (res->count > 0) {
> > +            ret = EINVAL;
>
> Actually it needs to be EMEDIUMTYPE I think.
> According to https://github.com/latchset/custodia/blob/master/API.md#deleting-keys
> We need to return:
>
> 406 not acceptable, type unknown/not permitted
>
> This is done in sec_errno_to_http_status():
>
>     case EMEDIUMTYPE:
>         return STATUS_406;
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

Checking the link you mentioned (thanks, btw) I do believe the proper
error to return is EEXIST, which will be translated to "409: if the
container is not empty."
Let me change this and re-submit the patch.